feat: background grid

epoll31 opened the pull request:
## Description

I added a simple component in the Effects group named BackgroundGrid that can be used to add a background grid to any page or any container very easily.

## Related Issue

Fixes #202 

## Proposed Changes

- added the BackgroundGrid component to `src/showcase/effects/BackgroundGrid.tsx`
- added the BackgroundGrid example to `src/showcase/ui-group/EffectCards.tsx`
  - for this, I added a check in the `data.map` that will display an Image only when `item.image` exists and will display a preview only when `item.preview` exists so that we can use components as previews instead of only images
- added the docs page to `src/app/(docs)/effects/background-grid/page.mdx`
  - I used `ComponentPreview` with the `preview` override prop so that I can customize the props like `size`, `color`, `strokeWidth`, etc. to make it fit better on the page.
- added the appropriate link to `src/components/Navigation.tsx` with the `new` tag
- added my name to the `credits.md` under this new addition

## Screenshots

<img width="271" alt="Screenshot 2024-05-21 at 10 47 58 AM" src="https://github.com/Ansub/SyntaxUI/assets/16091805/57a19ff9-09fc-4845-9322-7d9e8c3c782e">
<img width="296" alt="Screenshot 2024-05-21 at 10 48 11 AM" src="https://github.com/Ansub/SyntaxUI/assets/16091805/6ce7c58b-b971-4037-a73d-535f24db345b">

## Checklist

Please check the boxes that apply:

- [x] I have rebased my branch on top of the latest `main` branch.
- [x] I have tested the changes locally
- [x] I ran `npm run build` and build is successful
- [x] I have added necessary documentation (if applicable)
- [x] I have updated the credits.md file (if applicable)

Ansub commented on src/showcase/ui-group/EffectCards.tsx:
```diff
@@ -24,14 +38,17 @@ const EffectCards = () => {
+                    className="h-auto w-[150px] duration-300 ease-in-out group-hover:scale-110"
+                  />
+                )}
+                {item.preview}
```
Can you add a similar logic like it's been in `ComponentCards.tsx`

epoll31 commented:
I've made the changes, but the new `cn` function is breaking things, so it it not ready to be merged just yet.

epoll31 commented:
I updated the `BackgroundGrid` component to have an elliptical fade effect now. I also removed the fade controls from the component which makes it more opinionated.

<img width="1084" alt="Screenshot 2024-05-30 at 11 22 58 AM" src="https://github.com/SyntaxUI/syntaxui/assets/16091805/855ad72f-d4ee-4af4-b028-a6b5b7c2eacd">

Ansub commented:
can you fix the merge conflicts bro, i will merge it then

epoll31 commented:
Should be good now
